### PR TITLE
[audioplayers] Update audioplayers 

### PR DESCRIPTION
**Pull request by Swanseo0**
Dependency update:
- `audioplayers` from 1.0.1 to 4.1.0.
- `audioplayers_platform_interface` from 1.0.0 to 5.0.1.

Apply  the change caused by `audioplayers_platform_interface` update:
- Use `EventChannel` for audio event.
- Add `CreateAudioPlayer` and `DisposeAudioPlayer` methods.

Example app update.

### CODE REVIEW

File: packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc [hunk 1]
```diff
@@ -167,80 +229,98 @@ class AudioplayersTizenPlugin : public flutter::Plugin {
...
       return iter->second.get();
     }
+    return nullptr;
+  }
+
+  void CreateAudioPlayer(const std::string &player_id) {
+    auto event_channel = std::make_unique<FlEventChannel>(
+        registrar_->messenger(), "xyz.luan/audioplayers/events/" + player_id,
+        &flutter::StandardMethodCodec::GetInstance());
+    event_sinks_[player_id];
```
Comment (JSUYA) [inline packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc:239]: This is an unnecessary.

File: packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc [hunk 2]
```diff
@@ -167,80 +229,98 @@ class AudioplayersTizenPlugin : public flutter::Plugin {
...
-                                             int32_t duration) {
-      flutter::EncodableMap wrapped = {{flutter::EncodableValue("playerId"),
-                                        flutter::EncodableValue(player_id)},
-                                       {flutter::EncodableValue("value"),
-                                        flutter::EncodableValue(duration)}};
-      auto arguments = std::make_unique<flutter::EncodableValue>(wrapped);
-      channel->InvokeMethod("audio.onDuration", std::move(arguments));
+    event_channel->SetStreamHandler(
+        std::make_unique<AudioPlayerStreamHandler>(on_set_event_sink));
+    event_channels_[player_id] = std::move(event_channel);
```
Comment (JSUYA) [inline packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc:247]: The reason for using `event_channels_` is to maintain the lifecycle of unique_ptr? Is it used elsewhere?

File: packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc [hunk 3]
```diff
@@ -56,15 +74,40 @@ ReleaseMode StringToReleaseMode(std::string release_mode) {
   throw std::invalid_argument("Invalid release mode.");
 }
 
+#include <variant>
+using OnSetEventSink =
+    std::function<void(std::unique_ptr<FlEventSink> event_sink)>;
```
[the remark this answers is not in the document]
Comment (Swanseo0) [reply packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc]: changed it.

File: packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc [hunk 4]
```diff
@@ -167,80 +229,98 @@ class AudioplayersTizenPlugin : public flutter::Plugin {
...
-                                             const std::string &player_id,
-                                             int32_t duration) {
-      flutter::EncodableMap wrapped = {{flutter::EncodableValue("playerId"),
-                                        flutter::EncodableValue(player_id)},
-                                       {flutter::EncodableValue("value"),
-                                        flutter::EncodableValue(duration)}};
-      auto arguments = std::make_unique<flutter::EncodableValue>(wrapped);
-      channel->InvokeMethod("audio.onDuration", std::move(arguments));
+    event_channel->SetStreamHandler(
+        std::make_unique<AudioPlayerStreamHandler>(on_set_event_sink));
```
[the remark this answers is not in the document]
Comment (Swanseo0) [reply packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc]: Applied it. Thank you.

File: packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc (hunk 1 above)
Comment (Swanseo0) [reply packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc]: Deleted.

File: packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc (hunk 2 above)
Comment (Swanseo0) [reply packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc]: Deleted.

**Review by JSUYA — APPROVED** [review]
LGTM

**Pull request merged**